js/ui/text_box/action_button_collection/index.js

```diff
@@ -1,7 +1,60 @@
 import $ from "../../../core/renderer";
+import CustomButton from "./custom";
+import { extend } from "../../../core/utils/extend";
 import { find } from "../../../core/utils/array";
 
 const TEXTEDITOR_BUTTONS_CONTAINER_CLASS = "dx-texteditor-buttons-container";
+const TEXTEDITOR_BEFORE_BUTTONS_CONTAINER_CLASS = "dx-texteditor-before-buttons-container";
+const TEXTEDITOR_AFTER_BUTTONS_CONTAINER_CLASS = "dx-texteditor-after-buttons-container";
+
+function checkButtonInfo(buttonInfo) {
+    const checkButtonType = () => {
+        if(!buttonInfo || typeof buttonInfo !== "object" || Array.isArray(buttonInfo)) {
+            throw new TypeError("'buttons' option must include an object or a string items only");
```

Comment:
Should we use our `EXXX` errors message with links to the documentation?

Reply:
Yes, but we going to implement it in the separate PR